src/Playwright/Core/BrowserType.cs

```diff
@@ -149,8 +149,23 @@ public async Task<IBrowser> ConnectAsync(string wsEndpoint, BrowserTypeConnectOp
                 transport.Close("Disconnected");
                 throw new PlaywrightException("Malformed endpoint. Did you use launchServer method?");
             }
+            var browser = playwright.PreLaunchedBrowser;
+            browser.IsRemote = true;
+            browser.IsConnectedOverWebSocket = true;
+            browser.Disconnected += (_, e) => CloseBrowser(transport, browser);
+            return browser;
+        }
 
-            return playwright.PreLaunchedBrowser;
+        private void CloseBrowser(WebSocketTransport transport, Browser browser)
```

Comment:
Rename to be specific to the remote mode.

Reply:
Renamed it.